scripts/base/init-bare.zeek

```diff
@@ -942,6 +942,42 @@ type BacktraceElement: record {
 ## .. zeek:see:: backtrace print_backtrace
 type Backtrace: vector of BacktraceElement;
 
+## A hook that is invoked when an assert statement fails.
+##
+## If there's no hook handler installed that is using the break
+## statement, script execution continues even if an assert failed.
+## It is advisable to always have a low-priority handler just calling
+## break to avoid surprising behavior. Note that hook handler
+## execution is also short-circuited upon the first break - installing
+## multiple assertion_failure hooks requires coordination.
+##
+## cond: The string representation of the condition.
+##
+## msg: Message as string given to the assert statement.
+##
+## bt: Backtrace of the assertion error. The top element will contain
+##     the location of the assert statement that failed.
+##
+## .. zeek:see:: assertion_result
+type assertion_failure: hook(cond: string, msg: string, bt: Backtrace);
+
+## A hook that is invoked with the result of every assert statement.
```

Comment:
> I'm asking because if people started to use assert broadly to enforce pre-conditions, having such a hook implementation could become more expensive than one would expect. Might be useful to point that out in the docs; maybe even go so far and say that this is primarily for testing frameworks, not normal operation

Yep, will document that way. It was certainly not meant for normal operations, but without it it's difficult to judge if anything was executed at all. And yes, focus was mostly testing.